### PR TITLE
Feature/beta login logic

### PR DESCRIPTION
**Pull request by TzuHanLiang**
### DEVELOP

- [X] Description: Add Auth logic
  
#### Related Issues

- [X] Issue Number:  Link the related issue (e.g., #123)

#### Checklist

- [x] Used `@` in import paths.
- [x] Verified naming convention compliance ([Naming Convention Guidelines](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md)).
- [x] Coding style verification: checked
- [x] new Library: 0
- [x] new Class / Component: 2
    - [x] InformationStatement
    - [x] TermsOfServiceAndPrivacyPolicy
- [x] new loop: 0
- [x] new recursive: 0
- [x] high risk: 0
- [x] new sql: 0

#### UML Diagrams

- None

#### Additional Notes

- Ensured all links are up-to-date.
- Conducted thorough testing on responsive design changes.
- Documented any notable considerations or edge cases addressed.

### CODE REVIEW

File: src/components/login_page_body/login_page_body.beta.tsx [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
-    hasReadAgreement: boolean;
-  }>(APIName.LOGIN);
-  const { trigger: agreementTrigger } = APIHandler<null>(APIName.AGREEMENT);
 
-  const handleUserAgree = async (authWith: AuthWith) => {
+  const { trigger: agreementAPI } = APIHandler<null>(APIName.AGREE_TO_TERMS);
+
+  const handleUserAgree = async (provider: Provider) => {
     try {
       // 呼叫 API 紀錄用戶已同意條款
```
Comment (Luphia) [inline src/components/login_page_body/login_page_body.beta.tsx:29]: - Annotation format does not comply

File: src/components/login_page_body/login_page_body.beta.tsx [hunk 2]
```diff
@@ -43,78 +41,91 @@ const LoginPageBody = () => {
...
-
-      // 處理後端返回的 user 和 hasReadAgreement 資料
-      const { success, data } = response;
-      if (success && data) {
-        const { user, hasReadAgreement } = data;
-        if (!hasReadAgreement && !isAgreeWithInfomationConfirmModalVisible) {
-          // 如果用戶尚未同意條款，顯示同意條款的模態框
-          TOSNPrivacyPolicyConfirmModalCallbackHandler(() => handleUserAgree(authWith));
+  useEffect(() => {
+    // eslint-disable-next-line no-console
```
Comment (Luphia) [inline src/components/login_page_body/login_page_body.beta.tsx:45]: - The deprecation time needs to be annotated

File: src/components/login_page_body/login_page_body.beta.tsx [hunk 3]
```diff
@@ -43,78 +41,91 @@ const LoginPageBody = () => {
...
+    console.log('session:', session);
+    if (status === 'unauthenticated') {
+      try {
+        if (typeof update === 'function') {
+          update();
+        }
+      } catch (error) {
+        // eslint-disable-next-line no-console
```
Comment (Luphia) [inline src/components/login_page_body/login_page_body.beta.tsx:53]: - The deprecation time needs to be annotated

File: src/components/login_page_body/login_page_body.beta.tsx [hunk 4]
```diff
@@ -43,78 +41,91 @@ const LoginPageBody = () => {
...
+        console.error('Session update failed:', error);
+        // 可能需要跳轉到錯誤頁面或顯示錯誤消息
```
Comment (Luphia) [inline src/components/login_page_body/login_page_body.beta.tsx:55]: - Annotation format does not comply

File: src/components/login_page_body/login_page_body.beta.tsx [hunk 5]
```diff
@@ -43,78 +41,91 @@ const LoginPageBody = () => {
...
+      }
+    }
+    // eslint-disable-next-line no-console
```
Comment (Luphia) [inline src/components/login_page_body/login_page_body.beta.tsx:58]: - The deprecation time needs to be annotated

File: src/components/login_page_body/login_page_body.beta.tsx [hunk 6]
```diff
@@ -43,78 +41,91 @@ const LoginPageBody = () => {
...
+        if (isAgreeWithInfomationConfirmModalVisible) {
           agreeWithInfomationConfirmModalVisibilityHandler();
-        } else {
-          // 用戶已經同意條款，直接進行登入後的邏輯處理。TODO: (20240812-Tzuhan) route to select-company page
-          // eslint-disable-next-line no-console
-          console.log('用戶已登入:', user);
-          router.push(ISUNFA_ROUTE.SELECT_COMPANY);
         }
-      } else {
+        // 用户已同意條款，跳轉到公司選擇頁面
```
Comment (Luphia) [inline src/components/login_page_body/login_page_body.beta.tsx:68]: - The deprecation time needs to be annotated
- Annotation format does not comply

File: src/pages/api/auth/[...nextauth].ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+  providerAccountId,
+  name,
+  email,
+}: {
+  provider: string;
+  providerAccountId: string;
+  name: string;
+  email: string;
+}) => {
+  // eslint-disable-next-line no-console
```
Comment (Luphia) [inline src/pages/api/auth/[...nextauth].ts:58]: - The deprecation time needs to be annotated

File: src/pages/api/auth/[...nextauth].ts [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+           providerAccountId String?  @unique // OAuth 提供者帳號ID
+           hasReadAgreement Boolean  @default(false) // 用户是否同意了條款
+           createdAt        DateTime @default(now()) // 創建時間
+           updatedAt        DateTime @updatedAt // 更新時間
+           deletedAt        DateTime? // 删除時間（邏輯删除）
+         }
+       */
+      let newToken = token;
+      if (account && user) {
+        // 查找或創建新用户（不更新已有用户）
```
Comment (Luphia) [inline src/pages/api/auth/[...nextauth].ts:114]: - Annotation format does not comply

File: src/pages/api/auth/[...nextauth].ts [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+        });
+        newToken = {
+          ...token,
+          accessToken: account.access_token,
+          provider: account.provider,
+          userId: dbUser.id,
+          hasReadAgreement: dbUser.hasReadAgreement,
+        };
+      }
+      // eslint-disable-next-line no-console
```
Comment (Luphia) [inline src/pages/api/auth/[...nextauth].ts:129]: - The deprecation time needs to be annotated

File: src/pages/api/auth/error.ts [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  try {
+    const { error } = req.query;
+    // eslint-disable-next-line no-console
```
Comment (Luphia) [inline src/pages/api/auth/error.ts:6]: - The deprecation time needs to be annotated

File: src/pages/api/auth/error.ts [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+    console.error('Authentication error:', error);
+    return res.status(500).json({ message: `Authentication Error: ${error}` });
+  } catch (error) {
+    // eslint-disable-next-line no-console
```
Comment (Luphia) [inline src/pages/api/auth/error.ts:10]: - The deprecation time needs to be annotated

File: src/types/next-auth.d.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (Luphia) [inline src/types/next-auth.d.ts:1]: - The deprecation time needs to be annotated

File: src/pages/users/login-beta.tsx [hunk 13]
```diff
@@ -52,15 +54,44 @@ const LoginPage = () => {
...
+    return {
+      props: {
+        // session,
+        invitation: invitation as string,
+        action: action as string,
+        ...(await serverSideTranslations(locale as string, ['common'])),
+      },
+    };
+  } catch (error) {
+    // eslint-disable-next-line no-console
```
Comment (Luphia) [inline src/pages/users/login-beta.tsx:85]: - The deprecation time needs to be annotated

File: src/pages/users/login-beta.tsx [hunk 14]
```diff
@@ -52,15 +54,44 @@ const LoginPage = () => {
...
+    //     redirect: {
+    //       destination: ISUNFA_ROUTE.LOGIN_BETA,
+    //       permanent: false,
+    //     },
+    //   };
+    // }
+
+    return {
+      props: {
+        // session,
```
Comment (Luphia) [inline src/pages/users/login-beta.tsx:78]: - Annotation format does not comply

**Review by Luphia — CHANGES_REQUESTED** [review]

**Comment by TzuHanLiang** [comment]
- replaced by #2067

**Pull request closed without merging**